lib/AST/Builtins.cpp

```diff
@@ -491,8 +496,12 @@ namespace {
     }
 
     template <class G>
-    void addParameter(const G &generator) {
-      InterfaceParams.push_back(generator.build(*this, false));
+    void addParameter(const G &generator, bool isInOut = false) {
```

Comment:
I think it would be preferable to have two functions, `addParameter` and `addInOutParameter`, if that's feasible long-term (it looks like it is now). `bool`-typed arguments are always a code-smell for me. Alternatively you could convert this to your own booleanish enum with `NotInOut` and `IsInOut` or something else that is easy to distinguish from the first few characters.